From 21ebef6096da638f516cc4bd65d6eca030daaef2 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:11:14 -0400
Subject: [powerpc] Build little endian ppc64 kernel with ABIv2

Message-id: <1410545655-205645-446-git-send-email-dzickus@redhat.com>
Patchwork-id: 94397
O-Subject: [RHEL7 PATCH 445/626] powerpc: Build little endian ppc64 kernel with ABIv2
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 721aeaa9fdf35a672eef8ebdc4cd04bde38c3cea
Author: Anton Blanchard <anton@samba.org>
Date:   Mon Mar 10 21:06:12 2014 +1100

    powerpc: Build little endian ppc64 kernel with ABIv2

    Build the little endian ppc64 kernel with ABIv2 if the toolchain
    supports it. We can identify an ABIv2 capable toolchain by the
    -mabi=elfv2 compiler flag.

    Signed-off-by: Anton Blanchard <anton@samba.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/Makefile b/arch/powerpc/Makefile
index 81ca430..8784119 100644
--- a/arch/powerpc/Makefile
+++ b/arch/powerpc/Makefile
@@ -113,9 +113,13 @@ else
 endif
 endif
 
-CFLAGS-$(CONFIG_PPC64) := -mtraceback=no -mcall-aixdesc
-CFLAGS-$(CONFIG_PPC64) += $(call cc-option,-mabi=elfv1)
-AFLAGS-$(CONFIG_PPC64) += $(call cc-option,-mabi=elfv1)
+CFLAGS-$(CONFIG_PPC64) := -mtraceback=no
+ifeq ($(CONFIG_CPU_LITTLE_ENDIAN),y)
+CFLAGS-$(CONFIG_PPC64) += $(call cc-option,-mabi=elfv2,-mcall-aixdesc)
+AFLAGS-$(CONFIG_PPC64) += $(call cc-option,-mabi=elfv2)
+else
+CFLAGS-$(CONFIG_PPC64) += -mcall-aixdesc
+endif
 CFLAGS-$(CONFIG_PPC64) += $(call cc-option,-mcmodel=medium,-mminimal-toc)
 CFLAGS-$(CONFIG_PPC64) += $(call cc-option,-mno-pointers-to-nested-functions)
 CFLAGS-$(CONFIG_PPC32) := -ffixed-r2 $(MULTIPLEWORD)
-- 
1.7.1